src/Servers/Kestrel/samples/http2cat/Http2Utilities.cs

```diff
@@ -0,0 +1,920 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Buffers;
+using System.Buffers.Binary;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Pipelines;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Connections;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http;
+using Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http2;
+using Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http2.HPack;
+using Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Infrastructure;
+using Microsoft.Net.Http.Headers;
+
+namespace http2cat
+{
+    public class Http2Utilities : IHttpHeadersHandler
+    {
+        public static readonly int MaxRequestHeaderFieldSize = 16 * 1024;
+        public static readonly string _4kHeaderValue = new string('a', 4096);
+
+        public static readonly IEnumerable<KeyValuePair<string, string>> _browserRequestHeaders = new[]
```

Comment:
Anyway to share with this with our test infrastructure.

Reply:
Maybe. It's already diverged a fair bit from Http2TestBase, so I'm not sure it's worth the effort though.